doc/api-design.yml

```diff
@@ -0,0 +1,903 @@
+openapi: 3.0.3
+info:
+  title: Quizzard API
+  description: API documentation for the Quizzard application.
+  version: 1.0.0
+servers:
+  - url: http://localhost:8080/api
+    description: Development server
+components:
+  securitySchemes:
+    bearerAuth:
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+  schemas:
+    User:
+      type: object
+      properties:
+        id:
+          type: integer
+        username:
+          type: string
+        name:
+          type: string
+        email:
+          type: string
+          format: email
+        profile_picture:
+          type: string
+          format: uri
+        english_proficiency:
+          type: string
+          enum: [a1,a2,b1,b2,c1,c2]
+        points:
+          type: integer
+        created_quizzes:
+          type: integer
+        created_at:
+          type: string
+          format: date-time
+        updated_at:
+          type: string
+          format: date-time
+      required:
+        - username
+        - name
+        - email
+        - password
+        - english_proficiency
+    RegisterRequest:
+      type: object
+      properties:
+        username:
+          type: string
+        name:
+          type: string
+        email:
+          type: string
+          format: email
+        password:
+          type: string
+        english_proficiency:
+          type: string
+          enum: [a1,a2,b1,b2,c1,c2]
+      required:
+        - username
+        - name
+        - email
+        - password
+        - english_proficiency
+    LoginRequest:
+      type: object
+      properties:
+        username_or_email:
+          type: string
+        password:
+          type: string
+      required:
+        - username_or_email
+        - password
+    Quiz:
+      type: object
+      properties:
+        id:
+          type: integer
+        username:
+          type: string
+        title:
+          type: string
+        difficulty:
+          type: number
+          format: float
+        created_at:
+          type: string
+          format: date-time
+        updated_at:
+          type: string
+          format: date-time
+        solved:
+          type: boolean
+      required:
+        - user_id
+        - title
+        - difficulty
+    Question:
+      type: object
+      properties:
+        id:
+          type: integer
+        quiz_id:
+          type: integer
+        question_type:
+          type: string
+          enum: [english_to_turkish, turkish_to_english, english_to_sense]
+        word:
+          type: string
+        correct_answer:
+          type: string
+        wrong_answer1:
```

Comment:
Is there any specific reason we used individual fields for wrong answers rather than an array. I think they are equivalent but I wonder if there was a reason for choosing this approach?

Reply:
mysql database does not support lists and we store wrong answers in 3 different fields. If this API makes your work difficult we can pass them inside a list.